### PR TITLE
add mantlemint support

### PR DESCRIPTION
**Pull request by DeveloperInProgress**

### CODE REVIEW

File: packages/node-terra/src/indexer/apiterra.service.ts [hunk 1]
```diff
@@ -76,10 +89,54 @@ export class TerraClient {
...
+  async blockInfoMantlemint(height?: number): Promise<BlockInfo> {
+    const config = {
+      method: 'get',
+      url: `${this.mantlemintURL}/index/blocks/${height}`,
+    };
+    return axios(config)
+      .then((d) => d.data)
+      .catch((e) => {
+        throw e;
+      });
```
Comment (stwiname) [inline packages/node-terra/src/indexer/apiterra.service.ts:102]: We can use async await for these. (can also remove the eslint disable)

```suggestion
    const { data } = await axios(config);
    return data;
```

File: packages/node-terra/src/utils/terra-helper.ts [hunk 2]
```diff
@@ -102,10 +107,18 @@ export function filterEvents(
...
 
 async function getBlockByHeight(api: TerraClient, height: number) {
-  return api.blockInfo(height).catch((e) => {
+  let blockInfo: BlockInfo;
+  try {
+    if (api.mantlemintHealthOK) {
+      blockInfo = await api.blockInfoMantlemint(height);
+    } else {
+      blockInfo = await api.blockInfo(height);
+    }
```
Comment (stwiname) [inline packages/node-terra/src/utils/terra-helper.ts:116]: I think we can move this case to the TerraApi. This file shouldn't be concerned about the implementation that is used to fetch the content. We can do the same for TxInfo
 
```ts
// In TerraApi
async blockInfo(height) {
  if (this.mantlemintHealthOK) {
     return this.blockInfoMantlemint(height);
  }
  
  // return current implementation
}

```

File: packages/node-terra/src/utils/terra-helper.ts [hunk 3]
```diff
@@ -212,7 +225,16 @@ export async function fetchTerraBlocksBatches(
...
       }
-      const txInfos = await getTxInfobyHashes(api, txHashes);
+
+      let txInfos: TxInfo[];
+      if (api.mantlemintHealthOK) {
+        txInfos = await api.txsByHeightMantlemint(
+          blockInfo.block.header.height,
+        );
+      } else {
+        txInfos = await getTxInfobyHashes(api, txHashes);
```
Comment (stwiname) [inline packages/node-terra/src/utils/terra-helper.ts:235]: `getTxInfobyHashes` can be moved to the TerraApi

File: packages/node-terra/src/indexer/apiterra.service.ts [hunk 4]
```diff
@@ -76,10 +89,54 @@ export class TerraClient {
...
+    const health = axios(config)
+      .then((d) => d.data)
+      .catch((e) => {
+        throw e;
+      });
+    return health === 'OK';
+  }
+
+  // eslint-disable-next-line @typescript-eslint/require-await
+  async txsByHeightMantlemint(height: string): Promise<TxInfo[]> {
```
Comment (stwiname) [inline packages/node-terra/src/indexer/apiterra.service.ts:128]: How sure are we on this return type? Could there be differences in encoding? Or anything that needs to be parsed?

Comment (DeveloperInProgress) [reply packages/node-terra/src/indexer/apiterra.service.ts:128]: The returned object had the same set of keys as in TxInfo

**Review by stwiname — APPROVED** [review]

**Comment by stwiname** [comment]
@ianhe8x what are your thoughts on providing the mantlemint endpoint? Its currently just a CLI flag and is additional to the RPC endpoint

File: packages/node-terra/src/indexer/apiterra.service.ts [hunk 5]
```diff
@@ -77,19 +78,36 @@ export class TerraClient {
...
   async blockInfo(height?: number): Promise<BlockInfo> {
     if (this.mantlemintHealthOK && height) {
       return this.blockInfoMantlemint(height);
     }
-    return this.baseApi.tendermint.blockInfo(height, this.params);
+    const config = {
+      method: 'get',
+      url: height
+        ? `${this.tendermintURL}/cosmos/base/tendermint/v1beta1/blocks/${height}`
+        : `${this.tendermintURL}/cosmos/base/tendermint/v1beta1/blocks/latest`,
```
Comment (stwiname) [inline packages/node-terra/src/indexer/apiterra.service.ts:105]:
```suggestion
url: `${this.tendermintURL}/cosmos/base/tendermint/v1beta1/blocks/${height ?? 'latest'}`

```

Comment (DeveloperInProgress) [reply packages/node-terra/src/indexer/apiterra.service.ts:105]: looks good

**Pull request merged**